frontend/dispatcher.go

```diff
@@ -17,6 +17,47 @@ type HTTPError struct {
 	Message string
 }
 
+func db_requester() {
+	for {
+		db_request := <-db_open_channel
```

Comment:
```go
select {
case req := <-db_open_channel:
    ...
case <-done:
   //...cleanup
}
```

This means that db_requester will take a `chan` that will be used to indicate the program is shutting down which will let the DB you potentially just opened close all the way when it makes it back to the top of the for/select cycle.